experimental/incremental/url.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2020-2024 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package incremental
+
+import (
+	"net/url"
+	"strings"
+)
+
+// URLBuilder is a helper for building URLs.
+//
+// It is a simplified version of the interface of [net/url.URL].
+type URLBuilder struct {
```

Comment:
This doesn't seem like a real "builder" pattern. Instead, this seems like a `URI`, not a builder, and the `Build()` method could just as reasonably be `String()`.

Also, not sure this pays its freight -- it's extra surface area to maintain if/when we make it exported/non-experimental, and it doesn't seem hugely valuable.